File: bubble-sort/bubble-sort.ts
const bubbleSort = (arr: number[]) => {
  let [sorted, passes] = [false, 0]
  do {
    sorted = true
    for (let i = 0; i < arr.length - (1+passes); i++){
      const current = arr[i]
      if (current > arr[i + 1]){
        arr[i] = arr[i+1]
        arr[i + 1] = current
        sorted = false
      }
    }
    passes++
  } while ( sorted == false )
}

const nums = [100,99,1,4,8,5]

bubbleSort(nums)

console.log(nums)

File: bubble-sort/bubble-sort.py
def bubble_sort(alist):
  sorted, passes = False, 0
  while sorted == False:
    sorted = True
    for i, num in enumerate(alist[:len(alist) - (1+passes)]):
      if num > alist[i + 1]:
        alist[i], alist[i+1] = alist[i+1], num
        sorted = False

nums= [100,1,88,4,6,7,28]

bubble_sort(nums)

print(nums)